CONTRIBUTING.md

```diff
@@ -1,167 +1,113 @@
+* [Search](https://github.com/cdnjs/cdnjs/issues/) for possible duplicates before creating a new issue.
+* If you're the author/maintainer of a library, prepend `[author]` to the title when creating issues about that library. Such issues will be processed with higher priority than others.
+* Tag related [people](https://github.com/blog/821-mention-somebody-they-re-notified), [issues](https://github.com/blog/957-introducing-issue-mentions) and commits while creating a new issue to make collaboration easier.
 
- * a. [general conventions](#a-general-conventions)
+#### Popularity Requirements for New Libraries
 
- * b. [request a new library](#b-request-a-new-library)
+A library must satisfy the following criteria to be eligible for CDNJS:
+* 100 stars on GitHub **or** 500 downloads in the last month on npm
+* Should not be a personal hobby project
+* Must have at least one official and publicly accessible repository
+* Should be released under an [open source license](https://opensource.org/licenses)
 
- B. [pull request](#b-pull-request)
+### Pull Requests
 
- * a. [sparse checkout and shallow clone](#a-sparse-checkout-and-shallow-clone)
+#### Sparse Checkout and Shallow Clone
 
- * b. [common conventions](#b-common-conventions)
+There are several reasons why we recommend using a combination of sparse checkout and shallow clone to manage your local copy of the CDNJS repo:
 
- * c. [Updating an existing library](#c-updating-an-existing-library)
+* Cloning the whole repository to make a minor change or add a new library is not worthwhile.
+* The CDNJS repo on GitHub weighs around 2.7 GB and a local extracted copy would take up 70 GB of your disk space.
+* In case of case-insensitive file systems such as Windows and macOS, using sparse-checkout is a simple workaround to avoid the case issue (discussion here: [#3650](https://github.com/cdnjs/cdnjs/issues/3650)).
 
- * d. [adding a new library with its assets](#d-adding-a-new-library-with-its-assets)
+#### General Conventions for Pull Requests
 
- * e. [adding a new library by a single package.json](#e-adding-a-new-library-by-a-single-packagejson)
+1. CDNJS only hosts library with a production/stable release. Exceptions can be made if your pre-production version (alpha/beta/RC) is [popular](#popularity-requirements-for-new-libraries) and you'd like for it to be hosted on CDNJS.
+2. Rebase your CDNJS fork with upstream prior to every time you create a branch for a PR.
+3. Always edit on a branch. Never, ever make edits to the `master` branch. Always make a new branch before you edit files. We follow the [GitHub Flow](https://guides.github.com/introduction/flow/index.html) workflow.
 
- * f. [pre-flight checks](#f-pre-flight-checks)
+4. Don't make multiple unrelated changes in a single commit. Only strongly related changes must be included in a single commit.
+5. Every commit should be meaningful. For example, don't create a new commit to fix a typo you made in your previous commit.
```

Comment:
Maybe we can mention the `rebase` mechanism to let people not be afraid to use a new commit to fix the previous one.